### PR TITLE
[#IA] Grpc protocol

### PR DESCRIPTION
**Pull request by milica-okiljevic**

### CODE REVIEW

File: IntegrationAdaptersTests/ActionsAndBenefitsTests.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        public void Find_all_actions_and_benefits()
+        {
+            mock = new Mock<IActionsAndBenefitsRepository>();
+            mock.Setup(expression: t => t.GetAllPublishedActionsAndBenefitsMessages()).Returns(new List<ActionAndBenefitMessage>{
+                new ActionAndBenefitMessage { ActionID = guid, PharmacyName = "Jankovic", Text = "50% off brufen", DateFrom = new DateTime(2020, 5, 21), DateTo = new DateTime(2020, 6, 21) }
+            ,new ActionAndBenefitMessage { ActionID = new Guid("0e90a9a7-2722-45bb-a3de-7832131dcfe3"), PharmacyName = "BENU",Text = "2 vitamins Mg for 1 price", DateFrom= new DateTime(2020, 2, 16), DateTo = new DateTime(2020, 4, 12)} });
+
+            ActionsAndBenefitsService service = new ActionsAndBenefitsService(mock.Object);
+            service.GetAllPublishedActionsAndBenefitsMessages().Count.CompareTo(2);
+        }
```
Comment (milicasiriski) [inline IntegrationAdaptersTests/ActionsAndBenefitsTests.cs]: No assert method in this test

File: IntegrationAdaptersTests/ActionsAndBenefitsTests.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+using Xunit;
+using IntegrationAdapters.Services;
+
+namespace IntegrationAdaptersTests
+{
+    public class ActionsAndBenefitsTests
+    {
+        public Mock<IActionsAndBenefitsRepository> mock;
+        public Guid guid = new Guid("0e89a9a7-2722-45bb-a3de-7832131dcfe3");
+        public Guid guid2 = new Guid("a69b4d8e-55ce-4cdf-bfa6-7462e085a909");
```
Comment (milicasiriski) [inline IntegrationAdaptersTests/ActionsAndBenefitsTests.cs]: This is a good practice, to separate test objects that you use into fields, you should extend this and make ActionAndBenefitsMessage test object here and then use it in the tests - they will contain less code and be more readable

File: IntegrationAdaptersTests/ActionsAndBenefitsTests.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+        [Fact]
+        public void Find_existing_published_message()
+        {
+            mock = new Mock<IActionsAndBenefitsRepository>();
+            mock.Setup(expression: t => t.GetActionAndBenefitMessegeByID(It.IsAny<Guid>())).Returns( new ActionAndBenefitMessage { ActionID = guid, PharmacyName = "Jankovic", Text = "50% off brufen", DateFrom = new DateTime(2020,5,21) ,DateTo = new DateTime(2020,6,21)} );
```
Comment (milicasiriski) [inline IntegrationAdaptersTests/ActionsAndBenefitsTests.cs]: To continue on the previous comment, just instantiate this ActionAndBenefitsMessage in the field and use it in all the tests

File: IntegrationAdaptersTests/ActionsAndBenefitsTests.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            Assert.False(result);
+        }
+
+        [Fact]
+        public void Find_all_actions_and_benefits()
+        {
+            mock = new Mock<IActionsAndBenefitsRepository>();
+            mock.Setup(expression: t => t.GetAllPublishedActionsAndBenefitsMessages()).Returns(new List<ActionAndBenefitMessage>{
+                new ActionAndBenefitMessage { ActionID = guid, PharmacyName = "Jankovic", Text = "50% off brufen", DateFrom = new DateTime(2020, 5, 21), DateTo = new DateTime(2020, 6, 21) }
+            ,new ActionAndBenefitMessage { ActionID = new Guid("0e90a9a7-2722-45bb-a3de-7832131dcfe3"), PharmacyName = "BENU",Text = "2 vitamins Mg for 1 price", DateFrom= new DateTime(2020, 2, 16), DateTo = new DateTime(2020, 4, 12)} });
```
Comment (milicasiriski) [inline IntegrationAdaptersTests/ActionsAndBenefitsTests.cs]: And extract the other ActionsAndBenefitsMessage as well and you can just make the list here

Comment (milicasiriski) [reply IntegrationAdaptersTests/ActionsAndBenefitsTests.cs]: For example new List<ActionsAndBenefitsMessage> { actionsAndBenefits1, actionsAndBenefits1}

**Review by milicasiriski — CHANGES_REQUESTED** [review]
Good PR! Delete the .idea folder please 😄

**Review by dragana-todorovic — APPROVED** [review]

**Review by milicasiriski — APPROVED** [review]

**Pull request merged**